azurerm/internal/services/iothub/iothub_resource.go

```diff
@@ -346,6 +346,38 @@ func resourceArmIotHub() *schema.Resource {
 				},
 			},
 
+			"enrichment": {
+				Type: schema.TypeList,
+				// Currently only 10 enrichments is allowed for standard or basic tier, 2 for Free tier.
+				MaxItems:   10,
+				Optional:   true,
+				Computed:   true,
+				ConfigMode: schema.SchemaConfigModeAttr,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringMatch(
+								regexp.MustCompile("^[-_.a-zA-Z0-9]{1,64}$"),
+								"Enrichment Key name can only include alphanumeric characters, periods, underscores, hyphens, has a maximum length of 64 characters, and must be unique.",
+							),
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
```

Comment:
Can we validate this?

Reply:
According to [documentation](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-message-enrichments-overview), it could be:

> The value can be any of the following examples:
> 
> Any static string. Dynamic values such as conditions, logic, operations, and functions are not allowed. For example, if you develop a SaaS application that is used by several customers, you can assign an identifier to each customer and make that identifier available in the application. When the application runs, IoT Hub will stamp the device telemetry messages with the customer's identifier, making it possible to process the messages differently for each customer.
> 
> The name of the IoT hub sending the message. This value is $iothubname.
> 
> Information from the device twin, such as its path. Examples would be $twin.tags.field and $twin.tags.latitude.

So to me, as we can set any static string, no validation is needed

Reply:
Are empty strings allowed? if not we should validate its not empty

Reply:
You're right. 

Added ValidateFunc: validation.StringIsNotEmpty,